backend/de.metas.banking/de.metas.banking.base/src/main/java/de/metas/banking/payment/impl/PaySelectionUpdater.java

```diff
@@ -515,7 +575,25 @@ private void createOrUpdatePaySelectionLine(final PaySelectionLineCandidate cand
 		}
 	}
 
-	private void deletePaySelectionLine(@Nullable final I_C_PaySelectionLine paySelectionLine)
+	private @Nullable I_C_PaySelectionLine getPaySelectionLine(final @NonNull PaySelectionLineCandidate candidate)
+	{
+		final I_C_PaySelection paySelection = getC_PaySelection();
+
+		//
+		// Dequeue the existing C_PaySelectionLine to be updated.
+		// Also, make sure the existing pay selection is valid in our context.
+		final PaySelectionLinesToUpdate paySelectionLinesToUpdate = getPaySelectionLinesToUpdate();
+		final I_C_PaySelectionLine existingPaySelectionLine = paySelectionLinesToUpdate.dequeueFor(candidate).orElse(null);
+		if (existingPaySelectionLine != null && existingPaySelectionLine.getC_PaySelection_ID() != paySelection.getC_PaySelection_ID())
+		{
+			throw new AdempiereException("PaySelectionLine to update it's not matching the PaySelection header"
+					+ "\n Expected: " + paySelection
+					+ "\n Actual: " + existingPaySelectionLine.getC_PaySelection());
+		}
+		return existingPaySelectionLine;
+	}
+
+	private void dequePaySelectionLine(@Nullable final I_C_PaySelectionLine paySelectionLine)
```

Comment:
imho this is DELETE and not deque.

getPaySelectionLine shall be renamed ro deque because that's what it does.